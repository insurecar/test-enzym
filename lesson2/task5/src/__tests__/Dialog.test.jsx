import React from "react";
import { shallow } from "enzyme";
import Dialog from "../Dialog";
import "jsdom-global/register";

const mockFunction = jest.fn();

describe("Dialog", () => {
  test("should not be show <Dialog />", () => {
    const wrappedComponent = shallow(
      <Dialog onClose={mockFunction} isOpen={false} />
    );
    expect(wrappedComponent.find(".dialog").exists()).toBeFalsy();
  });

  test("should be show <Dialog />", () => {
    const wrappedComponent = shallow(<Dialog onClose={mockFunction} isOpen />);
    expect(wrappedComponent.find(".dialog").exists()).toBeTruthy();
  });

  test("should be closed <Dialog />", () => {
    const wrappedComponent = shallow(<Dialog onClose={mockFunction} isOpen />);
    wrappedComponent.find(".dialog__close-btn").simulate("click");
    expect(wrappedComponent.find(".dialog").exists()).toBeTruthy();
  });

  test("should be show h1", () => {
    const wrappedComponent = shallow(
      <Dialog onClose={mockFunction} isOpen title="Ivan" />
    );
    expect(wrappedComponent.find(".dialog__title").text()).toEqual("Ivan");
  });

  it("should be show children elements", () => {
    const wrappedComponent = shallow(
      <Dialog onClose={mockFunction} isOpen children="Grid Dynamics" />
    );
    expect(wrappedComponent.find(".dialog__content")).toMatchSnapshot();
  });
});
