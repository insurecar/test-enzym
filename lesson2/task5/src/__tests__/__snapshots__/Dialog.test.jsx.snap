// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should be show children elements 1`] = `
<div
  className="dialog__content"
>
  Grid Dynamics
</div>
`;
